kytos/core/helpers.py

```diff
@@ -332,3 +342,65 @@ def get_time(data=None):
     else:
         return None
     return date.replace(tzinfo=timezone.utc)
+
+
+def _read_from_filename(yml_file_path: Path) -> dict:
+    """Read from yml filename."""
+    spec_dict, _ = read_from_filename(yml_file_path)
+    return spec_dict
+
+
+def load_spec(yml_file_path: Path):
+    """Load and validate spec object given a yml file path."""
+    spec_dict = _read_from_filename(yml_file_path)
+    validate_spec(spec_dict)
+    return create_spec(spec_dict)
+
+
+def validate_openapi(spec):
+    """Decorator to validate a REST endpoint input.
+
+    Uses the schema defined in the openapi.yml file
+    to validate.
+    """
+
+    def validate_decorator(func):
+        @functools.wraps(func)
+        def wrapper_validate(*args, **kwargs):
+            try:
+                data = request.get_json()
```

Comment:
At this time, `validate_openapi` decorator won't be unit tested, it'll be covered with e2e since implementation details will also change, and it's notoriusly difficult to mock a flask.request object context on its own, so it's not worth doing it right now sinc eit'll be covered in this `2023.1` release anyway. In the future though, once we also have an openapi spec for core endpoints then we can use a mocked request client to unit test this.